internal/plugins/nfs-shares.go

```diff
@@ -340,13 +341,11 @@ func prometheusGetVector(api prom_v1.API, queryStr string) (model.Vector, error)
 		logg.Info("Prometheus query produced warning: %s", warning)
 	}
 	if err != nil {
-		//lint:ignore ST1005 Prometheus is a proper name
-		return nil, fmt.Errorf("Prometheus query failed: %s: %s", queryStr, err.Error())
+		return nil, fmt.Errorf("query failed for Prometheus: %s: %s", queryStr, err.Error())
```

Comment:
When I added the ling:ignore I was a bit cautious because I didn't know if this was used somewhere else. Not sure if we can assess if the error message is somewhere matched against.

Reply:
In this particular case, the error is not used anywhere else. The plugin interface provides a clear boundary between the plugin's internals and the rest of the application, so any matching would happen within the same file.

I don't like the wording though. It's not a query _for_ Prometheus (i.e. on behalf of Prometheus), it's a query _towards_ or _in_ Prometheus.

```suggestion
		return nil, fmt.Errorf("could not execute Prometheus query: %s: %s", queryStr, err.Error())
```